UPGRADE-6.7.md

```diff
@@ -326,36 +300,83 @@ The hidden radio input will no longer be in the HTML. The current page value wil
+
+## Custom field names and field set names validation
+Custom field names and field set names will be validated to not contain hyphens or dots, they must be valid Twig variable names (https://github.com/twigphp/Twig/blob/21df1ad7824ced2abcbd33863f04c6636674481f/src/Lexer.php#L46).
+Existing custom fields continue to work, however the validation will be enforced on new custom fields.
+
 ## Removal of deprecated properties of `CustomerDeletedEvent`
-* The deprecated properties `customerId`, `customerNumber`, `customerEmail`, `customerFirstName`, `customerLastName`, `customerCompany` and `customerSalutationId` of `CustomerDeleteEvent` will be removed and cannot be accessed anymore in a mail template when sending a mail via the `Checkout > Customer > Deleted` flow trigger.
-## Message queue size limit
+The deprecated properties `customerId`, `customerNumber`, `customerEmail`, `customerFirstName`, `customerLastName`, `customerCompany` and `customerSalutationId` of `CustomerDeleteEvent` will be removed and cannot be accessed anymore in a mail template when sending a mail via the `Checkout > Customer > Deleted` flow trigger.
 
-Any message queue message bigger than 256KB will be now rejected by default.
-To reduce the size of your messages you should only store the ID of an entity in the message and fetch it later in the message handler.
-This can be disabled again with:
+## Rule builder: Condition `customerDefaultPaymentMethod` removed
+* Removed condition `customerDefaultPaymentMethod` from rule builder, since customers do not have default payment methods anymore
+* Existing rules with this condition will be automatically migrated to the new condition `paymentMethod`, so the currently selected payment method
 
-```yaml
-shopware:
-    messenger:
-        enforce_message_size: false
+## Flow builder: Trigger `checkout.customer.changed-payment-method` removed
+* Removed trigger `checkout.customer.changed-payment-method` from flow builder, since customers do not have default payment methods anymore
+* Existing flows will be automatically disabled with Shopware 6.7 and removed in a future, destructive migration
+
+## Direct debit default payment: State change removed
+The default payment method "Direct debit" will no longer automatically change the order state to "in progress". Use the flow builder instead, if you want the same behavior.
+
+## New `technicalName` property for payment and shipping methods
+The `technicalName` property will be required for payment and shipping methods in the API.
+The `technical_name` column will be made non-nullable for the `payment_method` and `shipping_method` tables in the database.
+
+Plugin developers will be required to supply a `technicalName` for their payment and shipping methods.
+
+Merchants must review their custom created payment and shipping methods for the new `technicalName` property and update their methods through the administration accordingly.
 
+# API
+## Deletes by filter over the Sync API
```

Comment:
this was a regular feature in 6.6 🤔

Reply:
probably ended up under the wrong headline and should be part of the upgrade info and not major version change 🤔

Reply:
double checked, should be available already, removed the section